solr/bin/solr

```diff
@@ -2312,13 +2341,13 @@ function start_solr() {
 	echo ""
     fi
     # no lsof on cygwin though
-    if lsof -v 2>&1 | grep -q revision; then
+    if [ -n "$(get_port_tool)" ]; then
       echo -n "Waiting up to $SOLR_START_WAIT seconds to see Solr running on port $SOLR_PORT"
       # Launch in a subshell to show the spinner
       (loops=0
       while true
       do
-        running=$(lsof -t -PniTCP:$SOLR_PORT -sTCP:LISTEN || :)
+        running=$(check_port_in_use $SOLR_PORT)
```

Comment:
The ` || : ` in the initial command is curious. I infer that this protects against the possibility that a failed `lsof` command might still write to stdout, which would get captured into the `running` variable. Whatever the reason, might we want to either add ` || : ` to the `ss` and `netstat` variants of the port command, or move the ` || : ` back to the outer level here (ensuring that failed invocations of `lsof`/`ss`/`netstat` propagate failure exit codes via `check_port_in_use()`?)

Reply:
I must admint can't see that it would have had any effect here. But I may be coming it at from the wrong angle. I see it was originally added by @madrob https://github.com/apache/solr/commit/3ed17c2737db038c2908a1475099647fb4ef5b38  

If I create a script:
```
echo hello
exit 1
```
And call it with ` || :` the stdout still gets captured. The exit code will be 0 rather than 1, but I don't think that's adding anything here?

Reply:
ah yes, you're right! Maybe it just ensures that the script will proceed, regardless of how the script/environment does error handling (`set -e` etc.)?

Reply:
Yes, if lost doesn't exist we don't want the script to error out.

Reply:
> if lost doesn't exist

Sorry do you mean if `lsof` doesn't exist?
With these changes, `command -v` will already have been used to check if lsof / ss / netstat is available, and this won't be called if they aren't, it will already have branched to https://github.com/apache/solr/blob/dcb7d2406b574e24fa8636cf717a8af2039c79b4/solr/bin/solr#L2369